lib/prometheus_exporter/server/collector.rb

```diff
@@ -74,6 +76,11 @@ def register_metric_unsafe(obj)
       help = obj["help"]
       opts = symbolize_keys(obj["opts"] || {})
 
+      if !name
+        STDERR.puts "failed to register metric due to empty name #{obj}"
```

Comment:
this feels risky, we should not be printing to stderr, we have a logger can we use it?